### DISCUSSION TITLE
Prisma Schema Language and Curly Braces

### DISCUSSION THREAD

**Discussion by TScottJ** [prisma/prisma#20588] [Q&A] [answered]
### Question

Is there any compelling reason why the Prisma Schema Language (PSL) forces opening curly braces to be on the same line ([K&R style](https://en.wikipedia.org/wiki/Indentation_style#K&R_style))? For example, instead of:

**datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}**

I prefer to put the opening brace on a new line matched vertically with the closing brace, so that it is easier to read and less error prone ([Allman style](https://en.wikipedia.org/wiki/Indentation_style#Allman_style)):

**datasource db
{
  provider = "postgresql"
  url      = env("DATABASE_URL")
}**

...which of course works in all other languages that use the curly brace syntax (e.g. C/C++, C#, JavaScript, etc.), but causes a syntax error in PSL.

My intent is not to start a religious war over which style is better (that's been going on for 40 years already), rather I would just like to know why PSL forces K&R style and if there's any legitimate reason why it couldn't support both styles.

Thank you!

### Information about Prisma Schema, Client Queries and Environment (optional)

```js
// Add your schema.prisma and relevant query code
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by janpio** [comment, marked as answer]
Prisma started out as Graphcool, as Backend as a service for GraphQL. Later Prisma was born out of that, and even later Prisma Schema Language was evolved out of GraphQL. GraphQL uses `{...}` as well, and the `{` commonly lives in the same line as the "things before it": https://graphql.org/learn/schema/

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
